protocols/v2/codec-sv2/src/error.rs

```diff
@@ -1,69 +1,105 @@
+//! # Error Handling and Result Types
+//!
+//! This module defines error types and utilities for handling errors in the `codec_sv2` module.
+//! It includes the `Error` enum for representing various errors, a C-compatible `CError` enum for
+//! FFI, and a `Result` type alias for convenience.
+
 #[cfg(test)]
 use core::cmp;
 use core::fmt;
 use framing_sv2::Error as FramingError;
 #[cfg(feature = "noise_sv2")]
 use noise_sv2::{AeadError, Error as NoiseError};
 
+/// A type alias for results returned by the `codec_sv2` modules.
+///
+/// `Result` is a convenient wrapper around the `core::result::Result` type,
+/// using the `Error` enum defined in this crate as the error type.
 pub type Result<T> = core::result::Result<T, Error>;
 
+/// Enumeration of possible errors in the `codec_sv2` module.
+///
+/// This enum represents various errors that can occur within the `codec_sv2` module, including
+/// errors from related crates like [`binary_sv2`], [`framing_sv2`], and [`noise_sv2`].
 #[derive(Debug, PartialEq, Eq)]
 pub enum Error {
-    /// Errors from the `binary_sv2` crate
+    /// AEAD (`snow`) error in the Noise protocol.
+    #[cfg(feature = "noise_sv2")]
+    AeadError(AeadError),
+
+    /// Binary Sv2 data format error.
     BinarySv2Error(binary_sv2::Error),
+
+    /// Framing Sv2 error.
+    FramingError(FramingError),
+
+    /// Framing Sv2 error.
     FramingSv2Error(framing_sv2::Error),
-    /// Errors if there are missing bytes in the Noise protocol
-    MissingBytes(usize),
-    /// Errors from the `noise_sv2` crate
-    #[cfg(feature = "noise_sv2")]
-    NoiseSv2Error(NoiseError),
+
+    /// Invalid step for initiator in the Noise protocol.
     #[cfg(feature = "noise_sv2")]
-    AeadError(AeadError),
-    /// Error if Noise protocol state is not as expected
-    UnexpectedNoiseState,
+    InvalidStepForInitiator,
+
+    /// Invalid step for responder in the Noise protocol.
     #[cfg(feature = "noise_sv2")]
     InvalidStepForResponder,
+
+    /// Missing bytes in the Noise protocol.
+    MissingBytes(usize),
```

Comment:
Shouldn't this have a reference to Noise in its name? If it's just used to indicate missing bytes in the Noise protocol